js/battle.js

```diff
@@ -2107,8 +2107,9 @@ var Battle = (function () {
 		// 5 = seeking
 		this.playbackState = 0;
 
-		this.backdropImage = BattleBackdrops[0];
-		this.backdropImage = BattleBackdrops[Math.floor(Math.random() * BattleBackdrops.length)];
+		this.backdropImage = BattleBackdrops.gen6[0];
+		var gentext = (Tools.prefs('nopastgens'))? 'gen6' : 'gen' + this.gen;
+		this.backdropImage = BattleBackdrops[gentext][Math.floor(Math.random() * BattleBackdrops[gentext].length)];
```

Comment:
Oh, I didn't get you. I thought you meant the classic "if you use it twice, make a variable for it".